internal/campaigns/errors.go

```diff
@@ -0,0 +1,41 @@
+package campaigns
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sourcegraph/src-cli/internal/campaigns/graphql"
+)
+
+// unsupportedRepoSet provides a set to manage repositories that are on
+// unsupported code hosts. This type implements error to allow it to be
+// returned directly as an error value if needed.
+type unsupportedRepoSet map[*graphql.Repository]struct{}
+
+func (e unsupportedRepoSet) Error() string {
+	repos := []string{}
+	typeSet := map[string]struct{}{}
+	for repo := range e {
+		repos = append(repos, repo.Name)
+		typeSet[repo.ExternalRepository.ServiceType] = struct{}{}
+	}
+
+	types := []string{}
+	for t := range typeSet {
+		types = append(types, t)
+	}
+
+	return fmt.Sprintf(
+		"found repositories on unsupported code hosts: %s\nrepositories:\n\t%s",
+		strings.Join(types, ", "),
+		strings.Join(repos, ", "),
```

Comment:
as this list could grow a bit larger, maybe `"\n- "` would be a better separator?

Reply:
Blarg, that was `\n\t` before I started refactoring this out of an ad hoc structure in `Service`. Let me fix that real quick.